nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/TestCommentManagerImpl.java

```diff
@@ -15,19 +15,109 @@
  */
+@Features(CoreFeature.class)
 @Deploy("org.nuxeo.ecm.relations.api")
 @Deploy("org.nuxeo.ecm.relations")
 @Deploy("org.nuxeo.ecm.relations.jena")
 @Deploy("org.nuxeo.ecm.platform.comment.tests:OSGI-INF/comment-jena-contrib.xml")
 @Deploy("org.nuxeo.ecm.platform.comment.tests:OSGI-INF/relation-comment-manager-override.xml")
 public class TestCommentManagerImpl extends AbstractTestCommentManager {
 
+    public static final String QUERY_COMMENTS_AS_DOCUMENTS = "SELECT * FROM " + CommentsConstants.COMMENT_DOC_TYPE;
+
+    public static final String USERNAME = "Foo";
+
+    public static final String COMMENT_CONTENT_PROPERTY_XPATH = "comment:text";
+
+    public static final String COMMENT_CONTENT = "This is my comment";
+
+    @Inject
+    protected CoreSession systemSession;
+
+    @Inject
+    protected CoreFeature coreFeature;
+
+    @Inject
+    protected UserManager userManager;
+
+    @Test
+    public void testCreateReadDelete() {
+        CommentManager commentManager = new CommentManagerImpl(CommentServiceHelper.getCommentService().getConfig());
+        // Create a user
+        DocumentModel userModel = userManager.getBareUserModel();
+        String schemaName = userManager.getUserSchemaName();
+        userModel.setProperty(schemaName, "username", USERNAME);
+
+        // Create a folder
+        DocumentModel folder = systemSession.createDocumentModel("/", "folder", "Folder");
+        systemSession.createDocument(folder);
+
+        // Create a document
+        DocumentModel doc = systemSession.createDocumentModel("/folder/", "TestFile", "File");
+        doc = systemSession.createDocument(doc);
+
+        // Set The right ACE so a user can read the document
+        ACP acp = doc.getACP();
+        ACL acl = acp.getOrCreateACL(ACL.LOCAL_ACL);
+        acl.add(new ACE(USERNAME, SecurityConstants.READ, true));
+        doc.setACP(acp, true);
+
+        try (CloseableCoreSession userSession = coreFeature.openCoreSession(USERNAME)) {
+            // Get the document as the user
+            DocumentModel userDoc = userSession.getDocument(doc.getRef());
+
+            // Comment the document as the user
+            DocumentModel comment = userSession.createDocumentModel(CommentsConstants.COMMENT_DOC_TYPE);
+            comment.setPropertyValue(COMMENT_CONTENT_PROPERTY_XPATH, COMMENT_CONTENT);
+            commentManager.createComment(userDoc, comment);
+
+            // Check the comment document can be retrieved by a system session query
+            List<DocumentModel> dml = systemSession.query(QUERY_COMMENTS_AS_DOCUMENTS);
+            assertEquals(1, dml.size());
+
+            // Check the comment document cannot be retrieved by the user session query
+            dml = userSession.query(QUERY_COMMENTS_AS_DOCUMENTS);
+            assertEquals(0, dml.size());
+
+            // Check the comment can be retrieved by the user via the comment service
+            List<DocumentModel> comments = commentManager.getComments(userDoc);
```

Comment:
Could you use the new public `API` leveraging `Comment` interface? Service method is `getComments(CoreSession, String)`.

Note: `getComments(DocumentModel)` was kept because it was needed by `AnnotationService` for technical reason. This method should have been marked deprecated for public usage (but it's another story).